src/components/Select/Select.tsx

```diff
@@ -97,15 +105,46 @@ const Select = React.forwardRef<HTMLInputElement, Props & InputProps & TestProps
     const [open, setOpen] = React.useState(false);
     const inputRef = React.useRef<HTMLInputElement>(null);
     const combinedRefs = useCombinedRefs(inputRef, ref);
-    const [inputValue, setInputValue] = React.useState(defaultValue || selectedOption);
+
+    const initialValue = defaultValue || selectedOption;
+
+    const [inputValue, setInputValue] = React.useState(initialValue);
     const [searchValue, setSearchValue] = React.useState('');
 
+    const textFieldValue = searchValue || inputValue.label;
```

Comment:
Since `searchValue` and `inputValue.label` are both strings, if `searchValue === ''` (which is true on the initialization of searchValue) and let's say `inputValue.label === something with length >0`, then `searchValue ?? inputValue.label` would return `searchValue`, but `searchValue || inputValue.label` would return `inputValue.label` which is the correct one